lighthouse-core/gather/gather-runner.js

```diff
@@ -227,10 +224,13 @@ class GatherRunner {
             .then(_ => GatherRunner.beforePass(runOptions))
             .then(_ => GatherRunner.pass(runOptions))
             .then(_ => GatherRunner.afterPass(runOptions))
-            .then(loadData => {
-              // Merge pass trace and network data into tracingData.
-              config.trace && Object.assign(tracingData.traces, loadData.traces);
-              config.network && (tracingData.networkRecords = loadData.networkRecords);
+            .then(passData => {
+              // If requested by config, merge trace and network data for this
+              // pass into tracingData.
+              const passName = config.passName || Audit.DEFAULT_PASS;
+              config.trace && (tracingData.traces[passName] = passData.trace);
+              config.network && (tracingData.networkRecords[passName] = passData.networkRecords);
```

Comment:
in a readthough it's not clear that network is recorded but not saved. Should we rename the config property to `config.saveNetwork` or similar to clarify?